examples/MerminMagic/main.jl

```diff
@@ -1,85 +1,81 @@
 using Yao
-using Yao.EasyBuild, YaoPlots
-using Random
+
+# measurement operator table
+# see Table S1 in https://arxiv.org/abs/2206.12042
+op_mtx = [kron(I2, Z) kron(Z, I2) kron(Z, Z); kron(X, I2) kron(I2, X) kron(X, X); kron(-X, Z) kron(-Z, X) kron(Y, Y)]
+
+"""
+    bell_state(which::Int)::AbstractRegister
+
+Initialize bell state on a quantum register.
+
+# Arguments
+- 'which::Int': which of the four bell states to initialize
+  1 corresponds to ``(|00> + |11>)/√2``
+  2 corresponds to ``(|00> - |11>)/√2``
+  3 corresponds to ``(|01> + |10>)/√2``
+  4 corresponds to ``(|01> - |10>)/√2``
+"""
+function bell_state(which::Int)
+
+    @assert 1 <= which <= 4 "Input $(which) is not in range 1..4"
+    reg = zero_state(2)
+    apply!(reg, chain(put(2, 1 => H), cnot(2, 1, 2)))
+    (which == 2 || which == 4) && apply!(reg, put(2, 1 => Z))
+    (which == 3 || which == 4) && apply!(reg, put(2, 1 => X))
+    return reg
+end
 
 """
-    init_double_bell_state(reg::ArrayReg)
+    init_double_bell_state()::AbstractRegister
 
-Initialize two pairs of bell state.
+Initialize two pairs of bell state ``(|00> + |11>) ⊗ (|00> + |11>)/2``.
 
 First qubit of each bell state pair belongs to Alice and
 the others belong to Bob.
-
-# Arguments
-- 'reg::ArrayReg': the array of quantum register to initize state upon
 """
-function init_double_bell_state(reg::ArrayReg)
-    for k in [1, 3]
-        apply!(reg, put(4, k => H))
-        apply!(reg, cnot(4, k, k + 1))
-    end
+function init_double_bell_state()
+    return join(bell_state(1), bell_state(1))
 end
 
 
 """
-    meas_by_idx(reg::ArrayReg, i::Int, j::Int)
+    meas_by_idx(reg::AbstractRegister, i::Int, j::Int)::Vector{Float64}
 
 Perform Measurement based on index assigned by dealer.
 
 # Arguments
-- 'reg::ArrayReg': the array of quantum register to measure
+- `reg`: is the quantum register to measure
 - 'i::Int': the row idx assigned
 - 'j::Int': the column idx assigned
 
 """
-function meas_by_idx(reg::ArrayReg, i::Int, j::Int)
-    # measurement operator table
-    # see Table S1 in https://arxiv.org/abs/2206.12042
-    op_mtx = Matrix{KronBlock}(undef, 3, 3)
-    op_mtx = [kron(I2, Z) kron(Z, I2) kron(Z, Z); kron(X, I2) kron(I2, X) kron(X, X); kron(-X, Z) kron(-Z, X) kron(Y, Y)]
-
-    ans = zeros(Float64, 4)
-    for k in 1:2
-        # do measurement for Alice depending on the index given
-        @inbounds ans[k] = measure!(op_mtx[i, k], reg, (1, 3))[1]
-        # do measurement for Bob depending on the index given
-        @inbounds ans[k+2] = measure!(op_mtx[k, j], reg, (2, 4))[1]
-    end
+function meas_by_idx(reg::AbstractRegister, i::Int, j::Int)
 
-    return real.(ans)
+    alice_measure!(reg::AbstractRegister, row::Int) = [real(measure!(op, reg, (1, 3))) for op in op_mtx[row, :]]
+    bob_measure!(reg::AbstractRegister, col::Int) = [real(measure!(op, reg, (2, 4))) for op in op_mtx[:, col]]
+    return [alice_measure!(reg, i)'..., bob_measure!(reg, j)'...]
```

Comment:
This is very confusing to me.
1. You do not need a function called `meas_by_idx`. I am pretty sure a reader can not understand the output of this function by reading the docstring.
2. You should not concatenate the output, this is very confusing.
3. The prime in the return statement is useless.
4. You should not define `alice_measure!` and `bob_measure!` as closures, they should be regular functions.
5. You should not use `'argname'` even in docstring, because `'` does not have the meaning of string in Julia. You should quote the argname with `.

We definitely need a 4th round, but maybe it is much easier if we can go through the code together.